### DISCUSSION TITLE
Showing the rplidar A1 data

### DISCUSSION THREAD

**Discussion by leanhplus2612** [makerspet/support#15] [General]
Hi,
I am started with [lds_basic_esp32.ino](https://github.com/kaiaai/LDS/blob/main/examples/lds_basic_esp32/lds_basic_esp32.ino) example. In the serial monitor, It show 
"LiDAR model 3irobotics Delta-2G
LiDAR RX buffer size 1024, baud rate 115200
LiDAR start() result: OK"

My goal is print data getting from rplidar a1 and distance as well as angle? How can I do that?

**Comment by kaiaai** [comment]
Hello, 
  if you are getting `LiDAR model 3irobotics Delta-2G`, this means your sketch (somehow) instantiates a Delta-2G LiDAR. Probably using code like this:
```
#include <LDS_DELTA_2G.h>
...
LDS_DELTA_2G lidar;
```

You can switch to RPLidar A1 by changing the code like this:
```
#include <LDS_RPLIDAR_A1.h>
...
LDS_RPLIDAR_A1 lidar;
```

**Reply by leanhplus2612** [reply]
Hi [kaiaai](https://github.com/kaiaai),
Thank you for your advice! When I switched to RPLidar A1, I saw the motor was not working.
"LiDAR start() result: Device Info error
WARNING: is LDS connected to ESP32?"

However, I can received more information like
"Packet 5b, speed 0.00
1 0.00 256.17"
How can I fix it?

**Reply by leanhplus2612** [reply]
I have another question. The sketch have the 2 line below,
const uint8_t LDS_MOTOR_EN_PIN = 19; // ESP32 Dev Kit LiDAR enable pin
const uint8_t LDS_MOTOR_PWM_PIN = 15; // LiDAR motor speed control using PWM
what is LDS_MOTOR_EN_PIN do?

**Reply by makerspet** [reply]
LDS_MOTOR_EN pin is for (old) LiDAR models that have an enable pin, like YDLIDAR X2.

RPLidar A1 uses only TX/RX pins. So you can ignore that line of code.

"Device info error" means the library sends an ID and/or Start Motor commands to the RPlidar over the serial interface, but gets an unexpected serial response (or no response).

When the LiDAR motor is not rotating, the LiDAR still keeps sending data packets that contain the LiDAR speed - where the speed equals zero. So, it seems you are receiving some data over the serial port.

I'd recommend
- making sure that your RPlidar does receive serial data from the ESP32 - check which pin you have assigned to the ESP32 LiDAR Tx, check the TX signal with an oscilloscope if you have one, check the RPlidar connector, etc.
- once you are sure ESP32 TX (and RPLidar RX) work, try capturing ALL serial data from the Lidar (when you start the LiDAR) and posting a dump here. The dump should contain the serial response your RPLidar sends back when asked to start the motor.

**Reply by leanhplus2612** [reply]
Thanks for your advice, I still want to confirm some information:
1. TX and RX 's lidar pin can connect directly to TX, RX pin 's esp32?
2. When I choose Delta-2G LiDAR option in my code, motor is still working in spire of wrong type

**Reply by makerspet** [reply]
LiDAR Tx connects to ESP32 Rx, LiDAR Rx connects to ESP32 Tx. I use an ESP32 DevKitC v1 (30-pin) and connect to my RPLidar A1 directly since both have 3.3V I/O.
I suppose RPLidar A1 starts its motor on power-up automatically, unless it receives a serial command to stop its motor. I'd have to look at the code logic to make sense of it.

**Reply by leanhplus2612** [reply]
I have just used entire [lds_basic_esp32.ino](https://github.com/kaiaai/LDS/blob/main/examples/lds_basic_esp32/lds_basic_esp32.ino) with advice from @kaiaai.

**Comment by kaiaai** [comment]
Try disconnecting RPLidar A1 TX and RX. Turn on the power. Will the LiDAR start spinning?

**Reply by leanhplus2612** [reply]
Hello @kaiaai,
I think the advice 's @makerspet is quite useful. I used to see the status "OK" in serial monitor. However, Lidar didn't spin. Let me follow your recommendation and response later. I hope that I would still got your support.
Thank you!

**Comment by dawkoz** [comment]
Hi @leanhplus2612 , did you manage to get it working with RPLidar A1? If so, could you please share the code?

**Reply by leanhplus2612** [reply]
Hello @dawkoz, the library and example from LDS is working fine for me. My issue don't supply enough power for MCU.

**Reply by leanhplus2612** [reply]
BTW, thanks you for  @kaiaai  @makerspet your support !

**Reply by dawkoz** [reply]
@leanhplus2612 I tried to use lds_basic_esp32.ino with modifications mentioned in the conversation above, but it doesn't seem to detect the sensor. I changed the included header file and removed LDS_MOTOR_EN_PIN from the code. Could you share the code that worked for you, so that i can confirm that my problem is somewhere else? Thank you.
